### PR TITLE
bugfix: highlight enum cases correctly

### PR DESCRIPTION
**Pull request by kasiaMarek**
resolves: https://github.com/scalameta/metals/issues/5504

### CODE REVIEW

File: tests/cross/src/test/scala/tests/highlight/TypeDocumentHighlightSuite.scala [hunk 1]
```diff
@@ -198,4 +198,24 @@ class TypeDocumentHighlightSuite extends BaseDocumentHighlightSuite {
        |""".stripMargin,
   )
 
+  check(
```
Comment (jkciesluk) [inline tests/cross/src/test/scala/tests/highlight/TypeDocumentHighlightSuite.scala:201]: Can we move these tests to `Scala3DocumentHighlightSuite`?

**Review by jkciesluk — APPROVED** [review]
LGTM

File: tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala [hunk 2]
```diff
@@ -247,4 +247,24 @@ class Scala3DocumentHighlightSuite extends BaseDocumentHighlightSuite {
        |end extension""".stripMargin,
   )
 
+  check(
+    "enum-cases".tag(IgnoreScala2),
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:251]: You might want to ignore it for the nightlies for now, we should merge the change upstream to make it work there too. I can approve there as well

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala (hunk 2 above)
Comment (kasiaMarek) [reply tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:251]: should be good to go now

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**